File: Interviewbit/Tree/kth-smallest-element-in-tree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
 int ans=0,flag=1;
 void inorder(TreeNode* root, int &k, int &B){
    if(root==NULL)
        return;
    if(flag)
     inorder(root->left,k,B);
    k++;
    if(k==B){
        flag=0;
        ans=root->val;
        return ;
    }
    if(flag)
     inorder(root->right,k,B);
 }
 
int Solution::kthsmallest(TreeNode* A, int B) {
    int k=0;
    ans=0;
    inorder(A,k,B);
    return ans;
}

File: Interviewbit/Tree/kth-smallest-element-in-tree.java
/**
 * Definition for binary tree
 * class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) {
 *      val = x;
 *      left=null;
 *      right=null;
 *     }
 * }
 */
public class Solution {
    boolean found=false;
    int ans=0,k=0;
    private void solve(TreeNode root,int B){
        if(root==null) return;
        if(found==false)solve(root.left,B);
        k++;
        if(k==B){
            ans=root.val;
            found=true;
            return;
        }
        if(found==false)solve(root.right,B);
    }
    public int kthsmallest(TreeNode A, int B) {
        found=false;ans=0;k=0;
        if(A==null) return -1;
        solve(A,B);
        return ans;
    }
}
